homeassistant/components/blink/alarm_control_panel.py

```diff
@@ -32,55 +36,70 @@ async def async_setup_entry(
 
     sync_modules = []
     for sync_name, sync_module in data.sync.items():
-        sync_modules.append(BlinkSyncModule(data, sync_name, sync_module))
-    async_add_entities(sync_modules)
+        sync_modules.append(BlinkSyncModuleHA(data, sync_name, sync_module))
+    async_add_entities(sync_modules, update_before_add=True)
 
 
-class BlinkSyncModule(AlarmControlPanelEntity):
+class BlinkSyncModuleHA(AlarmControlPanelEntity):
     """Representation of a Blink Alarm Control Panel."""
 
     _attr_icon = ICON
     _attr_supported_features = AlarmControlPanelEntityFeature.ARM_AWAY
     _attr_name = None
     _attr_has_entity_name = True
 
-    def __init__(self, data, name, sync):
+    def __init__(self, data, name: str, sync) -> None:
         """Initialize the alarm control panel."""
-        self.data = data
+        self.data: Blink = data
         self.sync = sync
-        self._name = name
-        self._attr_unique_id = sync.serial
+        self._name: str = name
+        self._attr_unique_id: str = sync.serial
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, sync.serial)},
             name=f"{DOMAIN} {name}",
             manufacturer=DEFAULT_BRAND,
         )
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update the state of the device."""
         if self.data.check_if_ok_to_update():
             _LOGGER.debug(
                 "Initiating a blink.refresh() from BlinkSyncModule('%s') (%s)",
                 self._name,
                 self.data,
             )
-            self.data.refresh()
+            try:
+                await self.data.refresh(force=True)
+                self._attr_available = True
+            except asyncio.TimeoutError:
+                self._attr_available = False
+
             _LOGGER.info("Updating State of Blink Alarm Control Panel '%s'", self._name)
 
-        self._attr_state = (
-            STATE_ALARM_ARMED_AWAY if self.sync.arm else STATE_ALARM_DISARMED
-        )
         self.sync.attributes["network_info"] = self.data.networks
         self.sync.attributes["associated_cameras"] = list(self.sync.cameras)
         self.sync.attributes[ATTR_ATTRIBUTION] = DEFAULT_ATTRIBUTION
         self._attr_extra_state_attributes = self.sync.attributes
 
-    def alarm_disarm(self, code: str | None = None) -> None:
-        """Send disarm command."""
-        self.sync.arm = False
-        self.sync.refresh()
+    @property
+    def state(self) -> StateType:
+        """Return state of alarm."""
+        return STATE_ALARM_ARMED_AWAY if self.sync.arm else STATE_ALARM_DISARMED
 
-    def alarm_arm_away(self, code: str | None = None) -> None:
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        try:
+            await self.sync.async_arm(False)
+            await self.sync.refresh()
```

Comment:
It a refresh required, or could we use the return value of `await self.sync.async_arm(False)`?

Reply:
Need to go back to the mother ship to see if it actually made the change.  Otherwise we have to wait until the next polling update :(

Reply:
I probably do need the `force=True` here as well, good catch.